Mathlib/Analysis/Normed/Affine/Simplex.lean

```diff
@@ -0,0 +1,183 @@
+  · convert hr (e i) (e j) (e.injective.ne hij) using 2 <;> simp
+  · convert hr (e.symm i) (e.symm j) (e.symm.injective.ne hij) using 2
+
+/-- A simplex is regular if it is equivalent under an isometry to any reindexing. -/
+def Regular (s : Simplex R P n) : Prop :=
+  ∀ σ : Equiv.Perm (Fin (n + 1)), ∃ x : P ≃ᵢ P, s.points ∘ σ = x ∘ s.points
+
+@[simp] lemma regular_reindex_iff {s : Simplex R P m} (e : Fin (m + 1) ≃ Fin (n + 1)) :
+    (s.reindex e).Regular ↔ s.Regular := by
+  refine ⟨fun h σ ↦ ?_, fun h σ ↦ ?_⟩
+  · rcases h ((e.symm.trans σ).trans e) with ⟨x, hx⟩
+    refine ⟨x, ?_⟩
+    ext i
+    simpa using congrFun hx (e i)
+  · rcases h ((e.trans σ).trans e.symm) with ⟨x, hx⟩
+    refine ⟨x, ?_⟩
+    ext i
+    simpa using congrFun hx (e.symm i)
+
+lemma Regular.equilateral {s : Simplex R P n} (hr : s.Regular) : s.Equilateral := by
+  refine ⟨dist (s.points 0) (s.points 1), fun i j hij ↦ ?_⟩
+  have hn : n ≠ 0 := by omega
+  by_cases hi : i = 1
+  · rw [hi, dist_comm]
+    rcases hr (Equiv.swap 0 j) with ⟨x, hx⟩
+    nth_rw 2 [← x.dist_eq]
+    simp_rw [← Function.comp_apply (f := x), ← hx]
+    simp only [comp_apply, Equiv.swap_apply_left]
+    convert rfl
+    rw [Equiv.swap_apply_of_ne_of_ne (by norm_num [hn]) (by omega)]
+  · rcases hr ((Equiv.swap 0 i).trans (Equiv.swap 1 j)) with ⟨x, hx⟩
+    nth_rw 2 [← x.dist_eq]
+    simp_rw [← Function.comp_apply (f := x), ← hx]
+    simp only [Equiv.coe_trans, comp_apply, Equiv.swap_apply_left]
+    convert rfl
+    · exact Equiv.swap_apply_of_ne_of_ne hi hij
+    · rw [Equiv.swap_apply_of_ne_of_ne (by norm_num [hn]) (Ne.symm hi)]
+      simp
+
+end Simplex
+
+namespace Triangle
+
+lemma scalene_iff_dist_ne_and_dist_ne_and_dist_ne {t : Triangle R P} :
+    t.Scalene ↔ dist (t.points 0) (t.points 1) ≠ dist (t.points 0) (t.points 2) ∧
+      dist (t.points 0) (t.points 1) ≠ dist (t.points 1) (t.points 2) ∧
+      dist (t.points 0) (t.points 2) ≠ dist (t.points 1) (t.points 2) := by
+  refine ⟨fun h ↦
+    ⟨h.dist_ne (by decide : (0 : Fin 3) ≠ 1) (by decide : (0 : Fin 3) ≠ 2) (by decide) (by decide),
+     h.dist_ne (by decide : (0 : Fin 3) ≠ 1) (by decide : (1 : Fin 3) ≠ 2) (by decide) (by decide),
+     h.dist_ne (by decide : (0 : Fin 3) ≠ 2) (by decide : (1 : Fin 3) ≠ 2) (by decide) (by decide)⟩,
+    fun ⟨h₁, h₂, h₃⟩ ↦ ?_⟩
+  intro ⟨⟨x₁, x₂⟩, hx⟩ ⟨⟨y₁, y₂⟩, hy⟩ hxy
+  simp only at hx hy hxy
+  simp only [Subtype.mk.injEq, Prod.mk.injEq]
+  fin_cases x₁ <;> fin_cases x₂ <;> simp +decide only at hx <;>
+    fin_cases y₁ <;> fin_cases y₂ <;> simp +decide only at hy <;>
+    simp [h₁, h₂, h₃, h₁.symm, h₂.symm, h₃.symm] at hxy ⊢
+
+lemma equilateral_iff_dist_eq_and_dist_eq {t : Triangle R P} {i₁ i₂ i₃ : Fin 3} (h₁₂ : i₁ ≠ i₂)
+    (h₁₃ : i₁ ≠ i₃) (h₂₃ : i₂ ≠ i₃) :
+    t.Equilateral ↔ dist (t.points i₁) (t.points i₂) = dist (t.points i₁) (t.points i₃) ∧
+      dist (t.points i₁) (t.points i₂) = dist (t.points i₂) (t.points i₃) := by
```

Comment:
Could you please also provide a version that uses `0`, `1`, and `2 : Fin 3`?

Reply:
I've added the version with `0`, `1` and `2` in that order (this is an asymmetrical statement, any two equalities suffice out of three).